### PR TITLE
Update package data on sync

### PR DESCRIPTION
**Pull request by semtexzv**
We have some packages with no associations to advisories in CI

### CODE REVIEW

File: vmaas_sync/pkg_sync.go [hunk 1]
```diff
@@ -120,6 +120,8 @@ func storePackageDetails(tx *gorm.DB, advisoryIDs map[utils.Nevra]int, nameIDs m
 		})
 	}
 
-	tx = tx.Set("gorm:insert_option", "ON CONFLICT DO NOTHING")
-	return database.BulkInsertChunk(tx, toStore, chunkSize)
+	tx = database.OnConflictUpdateMulti(tx, []string{"name_id", "evra"},
+		"advisory_id", "summary_hash", "description_hash")
+
+	return database.BulkInsertChunk(tx.Debug(), toStore, chunkSize)
```
Comment (Josca) [inline vmaas_sync/pkg_sync.go:126]: forgotten .Debug()?

**Review by Josca — COMMENTED** [review]
feel free to merge it then

File: vmaas_sync/pkg_sync.go [hunk 2]
```diff
@@ -101,8 +101,27 @@ func storePackageStrings(tx *gorm.DB, data map[utils.Nevra]vmaas.PackagesRespons
...
+
+	var oldPackages []models.Package
+
+	database.Db.
+		Table("package p").
+		Select("pn.name, p.*").
+		Joins("JOIN package_name pn ON pn.id = p.name_id").
+		Where("p.evra in (?)", evras).
+		Where("pn.name in (?)", names).
+		Find(&oldPackages)
```
Comment (Josca) [inline vmaas_sync/pkg_sync.go:122]: missing error check here

File: vmaas_sync/pkg_sync.go [hunk 3]
```diff
@@ -111,15 +130,30 @@ func storePackageDetails(tx *gorm.DB, advisoryIDs map[utils.Nevra]int, nameIDs m
...
+		next := models.Package{
 			NameID:          nameIDs[nevra.Name],
 			EVRA:            nevra.EVRAString(),
 			DescriptionHash: desc[:],
 			SummaryHash:     sum[:],
 			AdvisoryID:      advisoryIDs[nevra],
-		})
+		}
+
+		for _, op := range oldPackages {
```
Comment (Josca) [inline vmaas_sync/pkg_sync.go:142]: seems to be a bit exhausting, right? What to load old packages to map[NameID-Evra]models.Package and then use it for old package - new one associations?

File: vmaas_sync/pkg_sync.go [hunk 4]
```diff
@@ -101,8 +101,27 @@ func storePackageStrings(tx *gorm.DB, data map[utils.Nevra]vmaas.PackagesRespons
 
 func storePackageDetails(tx *gorm.DB, advisoryIDs map[utils.Nevra]int, nameIDs map[string]int,
 	data map[utils.Nevra]vmaas.PackagesResponsePackageList) error {
-	toStore := make([]models.Package, 0, len(data))
+	inserts := make([]models.Package, 0, len(data))
+	updates := make([]models.Package, 0, len(data))
 
+	names := make([]string, 0, len(data))
+	evras := make([]string, 0, len(data))
```
Comment (Josca) [inline vmaas_sync/pkg_sync.go:108]: Is there any check whether data set is not empty?

**Review by Josca — CHANGES_REQUESTED** [review]

File: vmaas_sync/pkg_sync.go (hunk 4 above)
Comment (semtexzv) [reply vmaas_sync/pkg_sync.go:120]: There isn't one now, I think it shouldn't cause any problems.

**Review by Josca — APPROVED** [review]

**Pull request merged**